src/toolkits/object_detection/object_detector.cpp

```diff
@@ -335,16 +279,42 @@ size_t object_detector::get_version() const {
 }
 
 void object_detector::save_impl(oarchive& oarc) const {
-  if (training_model_) {
-    // If checkpointing during training, first copy weights from the backend.
-    synchronize_model(nn_spec_.get());
-  }
-  _save_impl(oarc, *nn_spec_, state);
+  Checkpoint* checkpoint = read_checkpoint();
```

Comment:
So to truly checkpoint would we also have to store the learning rate and current iterations to say resume training or does that work differently?

Reply:
Ohhh nvmd you also store the state. Got it.

Reply:
Hmm, but I don't pass the state to the backend. So I think you found a bug: resuming from a checkpoint shouldn't restart the learning-rate schedule.

I think the way to think about it is that the checkpoint should contain everything that the `DarknetYOLOModel` needs to operate correctly. The state is a higher-level concept, which is used to communicate with Python and to save to disk. That does imply that the state should somehow contain everything in the checkpoint (we explicitly handle the model weights separately)